cmd/abapEnvironmentRunAUnitTest.go

```diff
@@ -106,40 +104,49 @@ func runAbapEnvironmentRunAUnitTest(config *abapEnvironmentRunAUnitTestOptions,
 }
 
 func triggerAUnitrun(config abapEnvironmentRunAUnitTestOptions, details abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) (*http.Response, error) {
-	var aUnitConfigYamlFile []byte
+
 	abapEndpoint := details.URL
-	filelocation, err := filepath.Glob(config.AUnitConfig)
-	//Parse YAML AUnit run configuration as body for AUnit run trigger
-	if err == nil {
-		filename, err := filepath.Abs(filelocation[0])
-		if err == nil {
-			aUnitConfigYamlFile, err = ioutil.ReadFile(filename)
-		}
-	}
-	var AUnitConfig AUnitConfig
-	if err == nil {
-		var result []byte
-		result, err = yaml.YAMLToJSON(aUnitConfigYamlFile)
-		json.Unmarshal(result, &AUnitConfig)
-	}
-	var metadataString, optionsString, objectSetString string
-	if err == nil {
-		metadataString, optionsString, objectSetString, err = buildAUnitTestBody(AUnitConfig)
+	bodyString, err := buildAUnitRequestBody(config)
+	if err != nil {
+		return nil, err
 	}
 
 	//Trigger AUnit run
 	var resp *http.Response
-	var bodyString = `<?xml version="1.0" encoding="UTF-8"?>` + metadataString + optionsString + objectSetString
+
 	var body = []byte(bodyString)
-	if err == nil {
-		log.Entry().Debugf("Request Body: %s", bodyString)
-		details.URL = abapEndpoint + "/sap/bc/adt/api/abapunit/runs"
-		resp, err = runAUnit("POST", details, body, client)
-	}
-	if err != nil {
-		return resp, fmt.Errorf("Triggering AUnit test run failed: %w", err)
+	log.Entry().Debugf("Request Body: %s", bodyString)
+	details.URL = abapEndpoint + "/sap/bc/adt/api/abapunit/runs"
+	resp, err = runAUnit("POST", details, body, client)
+	return resp, err
+}
+
+func resolveAUnitConfiguration(config abapEnvironmentRunAUnitTestOptions) (aUnitConfig AUnitConfig, err error) {
+
+	if config.AUnitConfig != "" {
+		// Configuration defaults to AUnitConfig
+		result, err := abaputils.ReadConfigFile(config.AUnitConfig)
+		if err != nil {
+			return aUnitConfig, err
+		}
+		err = json.Unmarshal(result, &aUnitConfig)
+		return aUnitConfig, err
+
+	} else if config.Repositories != "" {
+		// Fallback / EasyMode is the Repositories configuration
+		repos, err := abaputils.GetRepositories((&abaputils.RepositoriesConfig{Repositories: config.Repositories}))
+		if err != nil {
+			return aUnitConfig, err
+		}
+		for _, repo := range repos {
+			aUnitConfig.ObjectSet.SoftwareComponents = append(aUnitConfig.ObjectSet.SoftwareComponents, SoftwareComponents{Name: repo.Name})
+		}
+		aUnitConfig.Title = "AUnit Test Run"
```

Comment:
Should we set the title to be fix when the User is using the `Repositories` flag?

Reply:
As of now it is required to provide a title in the aUnitConfig.yml. As there is no place for a title in the repositories.yml, a default has to be provided. Another option would be that the title is not mandatory anymore for both aUnitConfig.yml and repositories.yml.